File: 1476-count-negative-numbers-in-a-sorted-matrix/count-negative-numbers-in-a-sorted-matrix.java
class Solution {
    public int countNegatives(int[][] grid) {
        int row = grid.length;
        int count = 0;
        int col = grid[0].length;

        for(int i = 0; i < row; i++){
            for(int j = 0; j < col; j++){
                if(grid[i][j] < 0){
                    count++;
                }
            }
        }
        return count > 0 ? count : 0;
    }
}

File: 1476-count-negative-numbers-in-a-sorted-matrix/count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid):
        count = 0
        rows, cols = len(grid), len(grid[0])
        for i in range(rows):
            left, right = 0, cols - 1
            while left <= right:
                mid = (left + right) // 2
                if grid[i][mid] < 0:
                    right = mid - 1
                else:
                    left = mid + 1
            count += cols - left
        return count